### PR TITLE
add_channel must not overwrite existing channel

### PR DESCRIPTION
**Pull request by hackaugusto**

### CODE REVIEW

File: raiden/network/channelgraph.py [hunk 1]
```diff
@@ -236,21 +235,32 @@ def __ne__(self, other):
...
-            details.settle_timeout,
-        )
-        self.add_path(details.our_state.address, partner_state.address)
-
-        self.partneraddress_to_channel[partner_state.address] = channel
-        self.address_to_channel[channel_address] = channel
+
+        if details.our_state.address != self.our_address:
+            raise ValueError(
+                "Address mismatch, our_address doesn't match the  channel details"
```
Comment (LefterisJP) [inline raiden/network/channelgraph.py]: Remove extra space here: `match the  channel` between `the` and `channel`

File: raiden/network/channelgraph.py [hunk 2]
```diff
@@ -236,21 +235,32 @@ def __ne__(self, other):
...
-            self.token_address,
-            details.reveal_timeout,
-            details.settle_timeout,
-        )
-        self.add_path(details.our_state.address, partner_state.address)
-
-        self.partneraddress_to_channel[partner_state.address] = channel
-        self.address_to_channel[channel_address] = channel
+
+        if details.our_state.address != self.our_address:
```
Comment (LefterisJP) [inline raiden/network/channelgraph.py]: Shouldn't this code path be an assertion? If this happens it means that `add_channel` was called with completely wrong details and thus it's a programming error.

Comment (hackaugusto) [reply raiden/network/channelgraph.py]: I use assertion for code invariants, this is not an invariant but an invalid input

Comment (LefterisJP) [reply raiden/network/channelgraph.py]: Fair enough. Looks good apart from that. Thanks for the PR.

**Pull request merged**